### DISCUSSION TITLE
How to 24/7 record station with 3 mountpoints

### DISCUSSION THREAD

**Discussion by Ruben279** [AzuraCast/AzuraCast#6881] [Q&A]
Hello!

Just installed AzuraCast on a Linux VPS with Docker.
Everything is working fine.
I configured 1 station, with 3 mount points (two are Icecast connected, one is remote relay)

Can someone explain how to record these 3 mount points 24/7 and split them in seperate folders?

Thanks!

**Comment by LordHelmchen666** [comment]
Put this code in your Liquidsoap Custom Config. I put it at the end  in my config. Change recording_archive_path to match your install!
```
# Save Stream, open new file every hour
recording_archive_path = "/var/azuracast/stations/YOURSTATION/recordings/archive"
output.file(
%mp3(bitrate=128),
 {time.string("#{recording_archive_path}/%Y-%m-%d/%Y-%m-%d-T%H_%M_%S.mp3")}, radio, reopen_when={0m})
# Delete files in /recordings/archive older than 120 days and delete empty directorys
thread.when(every=3600., changed=false, { true },
    fun () -> list.iter(fun(msg) ->  log(msg, label="archive_cleaner"), 
       list.append(
            process.read.lines("find #{recording_archive_path}/* -type f -mtime +121 -delete"),
            process.read.lines("find #{recording_archive_path}/* -type d -empty -delete")
        )
    )
)
```
This will create a new mp3 file every hour and delete files older than 120 days.

This is how it looks. (I use aac for 24/7 recordings)
![Screenshot_20240121_151709](https://github.com/AzuraCast/AzuraCast/assets/43477041/1833a5f3-5d05-4756-b082-64603868057c)
![Screenshot_20240121_151841](https://github.com/AzuraCast/AzuraCast/assets/43477041/4aa8eabf-0fcf-4b7a-8b0d-815aa2eb41bd)

**Reply by Ruben279** [reply]
Thanks, tried that and recordings are being made now.
But the only thing hearable in the audio is "The station you are listening to is Azuracast....."

What I want, if the following situation. But I'm curious if Azuracast is the best way to go.

![Presentation1](https://github.com/AzuraCast/AzuraCast/assets/48621878/a125b0be-d257-4e50-8e60-738e804cbcdd)

**Reply by LordHelmchen666** [reply]
Are these three different streams? Or three DJs who should connect to the same stream?

**Comment by unudelaradio** [comment]
Hello! I find this very useful!

I tested today both the mp3 and aac formats. I know that aac at low bitrates is used for keeping a small size on disk and still a good audio quality. Are there other ways to encode in AAC format? For example, the code above can be configured for this format: AAC Plus (aacp) at 64kbps? 

Also my editing audio program only recognise FAAC.flt. The AAC codec in your code plays fine in audio players but won't open in my editing program (Cool Edit Pro). 

Maybe you can help me with some custom codes, both for:
- aacp (AAC Plus format)
- and that FAAC.flt (the only AAC format working in Cool Edit Pro).

Many thanks!

**Reply by LordHelmchen666** [reply]
As i understand this is aac+ already. Can you rename your file extension to "aacp"and try to open?
You can use many codecs. Here you can find  a list in LS documentation https://www.liquidsoap.info/doc-dev/encoding_formats.html

**Reply by unudelaradio** [reply]
Many thanks for the info! Very helpful! I've made some tests and i think the best quality when using low bitrates (64kbps) is to use the "mpeg2_he_aac". The sound is more stereophonic with "mpeg2_he_aac" and afterburner=true. 

Too bad that this AAC codec (mpeg2_he_aac) isn't listed in Azuracast mount points settings, for AutoDJ. I will suggest this.

One more question that i have. I saw that you reached somehow the Recordings folder via FTP or SFTP i guess. How can i do that? Is there an easy way to access Recordings folder via a FTP Client? In Azuracast i can configue SFTP users, but i can only see the Media folder when i'm logged in.

**Reply by LordHelmchen666** [reply]
I connect via SFTP with Dolphin as systemuser (not as Azuracast user).

**Reply by unudelaradio** [reply]
Hello! What am i doing wrong, because i customised this code and expected the files to be deleted after 1 or 2 days. I thought that by modifying the -mtime value (+2) will delete the files after 24 hours.

I also customised the file names, a little bit, and setted the archive path in media folder, because i'm not so familiar with Dolphin, and this was the only way to view and access the files.


```
# Save Stream, open new file every hour
recording_archive_path = "/var/azuracast/stations/brandingradio/media/Aircheck"
output.file(
%fdkaac(channels=2, samplerate=44100, bandwidth="auto", bitrate=64, afterburner=true, aot="mpeg2_he_aac"),
 {time.string("#{recording_archive_path}/%Y-%m-%d/%Y-%m-%d_-_T%H'%M'%S.aac")}, radio, reopen_when={0m})
# Delete files older than 2 days and delete empty directorys
thread.when(every=3600., changed=false, { true },
    fun () -> list.iter(fun(msg) ->  log(msg, label="archive_cleaner"), 
       list.append(
            process.read.lines("find #{recording_archive_path}/* -type f -mtime +2 -delete"),
            process.read.lines("find #{recording_archive_path}/* -type d -empty -delete")
        )
    )
)
```
